decoder.rs

```diff
@@ -73,7 +74,8 @@ pub trait Decoder {
     fn decode_f64(&mut self) -> Result<f64, DecodingError>;
 }
 
-impl Decoder for Cursor<&mut Vec<u8>> {
+#[duplicate_item(cursor; [Cursor<&mut Vec<u8>>]; [Cursor<Vec<u8>>])]
+impl Decoder for cursor {
```

Comment:
Could possibly avoid the use of this external dependency for generating multiple implementations with a generic trick
```
impl<'a, T> Decoder for Cursor<T>
where
    T: AsRef<[u8]>,
{
```
This should satisfy both `Cursor<&mut Vec<u8>>` and `Cursor<Vec<u8>>`

Reply:
Cool! Fixed!